controller/src/main/java/com/softserve/borda/controllers/BoardController.java

```diff
@@ -349,35 +350,56 @@ public Board getBoard(@PathVariable Long id) {
         return new ResponseEntity<>(tagDTO, HttpStatus.OK);
     }
 
-    @PostMapping("/boards/{boardId}/lists/{listId}/tickets/{ticketId}/tags")
+    @PostMapping("/boards/{boardId}/tags")
+    public ResponseEntity<TagDTO> addNewTagToBoard
+            (@RequestBody TagDTO tagDTO,
+             @PathVariable Long boardId) {
+        try {
+            if (Objects.isNull(tagDTO.getBoardId())) {
+                tagDTO.setBoardId(boardId);
+            }
+            Tag tag = modelMapper.map(tagDTO, Tag.class);
+            tag = tagService.create(tag);
+            tagDTO = modelMapper.map(tag, TagDTO.class);
+
+            return new ResponseEntity<>(tagDTO, HttpStatus.OK);
+        } catch (Exception e) {
+            log.severe(e.getMessage());
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @PostMapping("/boards/{boardId}/lists/{listId}/tickets/{ticketId}/tags/{tagId}")
     public ResponseEntity<TicketDTO> addTagToTicket
             (@PathVariable Long ticketId,
-             @RequestBody TagDTO tagDTO,
+             @PathVariable Long tagId,
              @PathVariable String boardId,
              @PathVariable String listId) {
-        // TODO rework tags to relate to certain board
-        //  and here tags should not be created, just added to ticket
-        Tag tag = modelMapper.map(tagDTO, Tag.class);
-        tag = tagService.create(tag);
-        Ticket ticket = ticketService.addTagToTicket(ticketId, tag.getId());
+        Ticket ticket = ticketService.addTagToTicket(ticketId, tagId);
         TicketDTO ticketDTO = modelMapper.map(ticket, TicketDTO.class);
 
         return new ResponseEntity<>(ticketDTO, HttpStatus.OK);
     }
 
     @GetMapping("/boards/{boardId}/tags")
     public ResponseEntity<List<TagDTO>> getAllTagsForBoard
-            (@PathVariable String boardId) {
-        // TODO rework tags to relate to certain board
-        //  and make this return all tags of one board
-        return new ResponseEntity<>(HttpStatus.OK);
+            (@PathVariable Long boardId) {
+        List<Tag> tags = tagService.getAllTagsByBoardId(boardId);
+        List<TagDTO> tagDTOs = tags.stream().map(tag ->
+                modelMapper.map(tag, TagDTO.class))
+                .collect(Collectors.toList());
+
+        return new ResponseEntity<>(tagDTOs, HttpStatus.OK);
     }
 
     @PutMapping("/boards/{boardId}/tags/{tagId}")
     public ResponseEntity<TagDTO> updateTagByTagId
             (@PathVariable Long tagId,
              @RequestBody TagDTO tagDTO,
-             @PathVariable String boardId) {
+             @PathVariable Long boardId) {
+        if (Objects.isNull(tagDTO.getBoardId())) {
```

Comment:
I think, it is redundant check.
And in general, it better to create `UpdateTagDTO` that contains properties that are available for update and then set `boardId` what you get from path variable.

Reply:
Will change, thanks for advice